### PR TITLE
feat: set chunk size to model context size for LLM chains

### LINKED ISSUE #35: Make summary chain aware of model's context size
The summary chain currently cuts its input in chunks and recursively summarizes them until a certain lower size limit is reached.
This is because model context sizes were historically very small. Recently this has changed however.

To make use of these larger context sizes and to increase the quality of our summaries, we should make the summary chain aware of the context size capacity of the currently selected model and adjust the chunk size to it.

### Code pointers

- Chunk size: https://github.com/nextcloud/llm2/blob/main/lib/summarize.py#L75
- Context size config: https://github.com/nextcloud/llm2/blob/main/default_config/config.json#L46
- config loading: https://github.com/nextcloud/llm2/blob/main/lib/chains.py#L21-L40

### PR DESCRIPTION
**Pull request by edward-ly**
Closes #35. I also went ahead and made similar changes to the other chains even if they don't seem to be used at the moment, but if you don't need it for the other chains, then you can just merge the first commit.

### CODE REVIEW

File: lib/chains.py [hunk 1]
```diff
@@ -82,15 +82,16 @@ def generate_chains():
...
-    chains[model_name + ":core:text2text:summary"] = lambda: SummarizeChain(llm_chain=llm_chain())
+    chains[model_name + ":core:text2text:summary"] = lambda: SummarizeChain(llm_chain=llm_chain(), chunk_size=n_ctx)
     chains[model_name + ":core:text2text:headline"] = lambda: HeadlineChain(llm_chain=llm_chain())
     chains[model_name + ":core:text2text:topics"] = lambda: TopicsChain(llm_chain=llm_chain())
-    # chains[model_name + ":core:text2text:simplification"] = lambda: SimplifyChain(llm_chain=llm_chain())
-    # chains[model_name + ":core:text2text:formalization"] = lambda: FormalizeChain(llm_chain=llm_chain())
-    # chains[model_name + ":core:text2text:reformulation"] = lambda: ReformulateChain(llm_chain=llm_chain())
+    # chains[model_name + ":core:text2text:simplification"] = lambda: SimplifyChain(llm_chain=llm_chain(), chunk_size=n_ctx)
+    # chains[model_name + ":core:text2text:formalization"] = lambda: FormalizeChain(llm_chain=llm_chain(), chunk_size=n_ctx)
+    # chains[model_name + ":core:text2text:reformulation"] = lambda: ReformulateChain(llm_chain=llm_chain(), chunk_size=n_ctx)
```
Comment (marcelklehr) [inline lib/chains.py:95]: Awesome, can we set this to 70% of the the context size so that there's still room for the response tokens?

Comment (edward-ly) [reply lib/chains.py]: Done. Ideally, I'd like the percentage to be adjustable somehow, but I'm not sure if a user-adjustable setting would make sense either.

Comment (marcelklehr) [reply lib/chains.py]: We could also subtract the target summary size from the context size

Comment (edward-ly) [reply lib/chains.py]: By "target summary size", I'm guessing you're referring to `max(len(inputs['input']) * 0.2, 1000)` in `summarize.py`?

Comment (marcelklehr) [reply lib/chains.py]: Yeah

Comment (edward-ly) [reply lib/chains.py]: Done, but only on the summarize chain; I'm not sure if the other chains should behave similarly. Also, I made sure there is a minimum (positive) chunk size, which I temporarily set to 2048, but I'm not sure if that's an appropriate value either.

**Review by marcelklehr — APPROVED** [review]
Seems like a good default. Thank you!

**Pull request merged**